### PR TITLE
🎉 New Source: Google Mobility Reports [low-code cdk]

### PR DESCRIPTION
**Pull request by vincentkoc**
## What
Resolves:
* https://github.com/airbytehq/connector-contest/issues/178
* https://github.com/airbytehq/airbyte/issues/18329

## How
Using low-code cdk I added one of the streams:
* Global report

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-mobility-reports/integration_tests/catalog.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": {
+          "country_region_code": {
+            "type": "string"
+          },
+          "country_region": {
+            "type": "string"
+          },
+          "sub_region_1": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-mobility-reports/integration_tests/catalog.json:16]: Remove this file isn't used.

File: airbyte-integrations/connectors/source-google-mobility-reports/source_google_mobility_reports/google_mobility_reports.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  global_mobility_report_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "global_mobility_report"
+      primary_key: ["placeid","date"]
+      path: "/Global_Mobility_Report.csv"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-mobility-reports/source_google_mobility_reports/google_mobility_reports.yaml:25]: Not sure if this connector should be build using the low-code.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
@koconder please check my comments, not sure if this connector is the best option using the low-code. Maybe other option is to allow users to insert the Path to download the report based in a region.

image

**Comment by marcosmarxm** [comment]
@koconder can you start sharing the local integration tests?

**Comment by marcosmarxm** [comment]
Please ask my review again when all comments are resolved.

**Comment by marcosmarxm** [comment]
@koconder did you have time to check my comment? Feel free to request my review again.

**Comment by marcosmarxm** [comment]
@koconder are you still working on this?

**Comment by koconder** [comment]
@marcosmarxm I’ll get onto the review items in the next day or two!

File: airbyte-integrations/connectors/source-google-mobility-reports/source_google_mobility_reports/google_mobility_reports.yaml (hunk 2 above)
Comment (koconder) [reply airbyte-integrations/connectors/source-google-mobility-reports/source_google_mobility_reports/google_mobility_reports.yaml:25]: @marcosmarxm its a single data-set, so i opted for a simple single stream of data versus providing the filtering. the other filtered data is in PDF and not CSV

**Comment by marcosmarxm** [comment]
Tests are not passing: `./gradlew airbyte-integrations:connectors:source-google-mobility-report:integrationTest`

**Comment by marcosmarxm** [comment]
Removing this connector from Hacktoberfest event. Tests are not passing and the output of the connector doesn't follow one premise of alpha low-code cdk which responses must be JSON objects.
https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview/#what-connectors-can-i-build-using-the-low-code-framework

@koconder you can continue the work and implement this connector in your pace.

**Pull request closed without merging**